editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/main/MainFragment.kt

```diff
@@ -75,6 +81,28 @@ class MainFragment : Fragment() {
             activity!!.invalidateOptionsMenu()
         }
 
+        // Button to add a new flag
+        val addFlag = view.findViewById<View>(R.id.addButton)
+        addFlag.setOnClickListener {
+
+            //Add flag to centre of screen
+
+            val relativeLayout = RelativeLayout(this.context)
+
+            val layoutParams = RelativeLayout.LayoutParams(
+                    RelativeLayout.LayoutParams.MATCH_PARENT,
+                    RelativeLayout.LayoutParams.MATCH_PARENT)
+            layoutParams.addRule(RelativeLayout.CENTER_IN_PARENT)
```

Comment:
You can use a [`ConstraintSet`](https://developer.android.com/reference/android/support/constraint/ConstraintSet) to help with positioning the image.

Using the constraint layout, you can place the flag in the center of the screen and it will account for the size so it appears in the correct place.

Afterwards, you can adjust its [translation, aka x and y offset](https://developer.android.com/reference/android/support/constraint/ConstraintSet#settranslation).